File: typescript/src/bad/badProcessor.ts
import { badFilter } from './badFilter';
import { badAggregator } from './badAggregator';
import { badCounter } from './badCounter';
import { badEnricher } from './badEnricher';
import { badLogger } from './badLogger';
import { results, counter } from './badGlobals';

export function badProcessor(
    items: string[],
    flag: boolean,
    knownItems: Set<string>,
    add1: string[],
    add2: Set<string>,
    log: boolean,
    user: string
): Map<string, number> {
    const filtered = badFilter(items, knownItems, flag);
    badAggregator(filtered, results);
    badCounter(filtered);
    badLogger(filtered, log);
    badEnricher(add1, add2, flag, user);
    return results;
}

File: java/javaproj/src/main/java/org/codereviewpractice/bad/BadProcessor.java
package org.codereviewpractice.bad;

import java.util.*;

public class BadProcessor {

    private final BadFilter f = new BadFilter();
    private final BadAggregator a = new BadAggregator();
    private final BadCounter c = new BadCounter();
    private final BadEnricher e = new BadEnricher();

    public Map<String, Integer> process(
            List<String> l,
            boolean flag,
            Set<String> known,
            List<String> add1,
            Set<String> add2,
            boolean log,
            String user
    ) {
        List<String> filtered = f.doFilter(l, known, flag);
        a.add(filtered, BadGlobals.results);
        c.count(filtered);
        maybeLog(filtered, log); // now part of this class
        e.doStuff(add1, add2, flag, user);
        return BadGlobals.results;
    }

    private void maybeLog(List<String> stuff, boolean should) {
        if (should == true) {
            System.out.println("Did stuff with: " + stuff);
        }
    }
}
